packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -214,19 +215,27 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       onRenderMenuIcon = this._onRenderMenuIcon
     } = props;
 
-    return React.createElement(
-      tag,
-      buttonProps,
-      React.createElement(
-        'div' as any,
-        { className: this._classNames.flexContainer },
-        onRenderIcon(props, this._onRenderIcon),
-        this._onRenderTextContents(),
-        onRenderAriaDescription(props, this._onRenderAriaDescription),
-        onRenderChildren(props, this._onRenderChildren),
-        !this._isSplitButton && (menuProps || menuIconProps || this.props.onRenderMenuIcon) && onRenderMenuIcon(this.props, this._onRenderMenuIcon),
-        this.state.menuProps && onRenderMenu(menuProps, this._onRenderMenu)
-      ));
+    const Content = <Tag {...buttonProps }>
+      <div className={ this._classNames.flexContainer } >
+        { onRenderIcon(props, this._onRenderIcon) }
+        { this._onRenderTextContents() }
+        { onRenderAriaDescription(props, this._onRenderAriaDescription) }
+        { onRenderChildren(props, this._onRenderChildren) }
+        { !this._isSplitButton && (menuProps || menuIconProps || this.props.onRenderMenuIcon) && onRenderMenuIcon(this.props, this._onRenderMenuIcon) }
+        { this.state.menuProps && !this.state.menuProps.doNotLayer && onRenderMenu(menuProps, this._onRenderMenu) }
+      </div>
+    </Tag>
+
+    if (menuProps && menuProps.doNotLayer) {
+      return (
+        <span style={ { display: 'inline-block' } }>
+          { Content }
```

Comment:
Technically in HTML a div can't be inside of a span which happens in this case

Reply:
ah, I can go with a div then